spring-ai-core/src/main/java/org/springframework/ai/vectorstore/VectorStore.java

```diff
@@ -5,6 +5,10 @@
 
 import org.springframework.ai.document.Document;
 import org.springframework.ai.document.DocumentWriter;
+import org.springframework.ai.vectorstore.filter.Filter;
+import org.springframework.ai.vectorstore.filter.FilterExpressionBuilder;
+import org.springframework.ai.vectorstore.filter.FilterExpressionTextParser;
+import org.springframework.ai.vectorstore.filter.converter.PrintFilterExpressionConverter;
 
 public interface VectorStore extends DocumentWriter {
```

Comment:
wondering how to expose arguments to search which are the antlr based filter string or the filter dsl expression directly

Reply:
done, check the javadoc for both search with filters methods